src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -121,12 +131,63 @@ function useChatUtilsHook(): UseChatUtilsHook {
     });
   };
 
+  const helperSaveChannelDetail = async (channel: ChannelList, response: any) => {
+    if (!localDb) return;
+    const {originalMembers} = getChannelListInfo(response?.channel);
+
+    const promise = originalMembers?.map((member) => {
+      return new Promise((resolve, reject) => {
+        const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+        try {
+          user.saveOrUpdateIfExists(localDb).then(() => resolve(null));
+        } catch (e) {
+          reject(e);
+        }
+      });
+    });
+
+    await Promise.all(promise).catch((e) => console.log('saveChannelDetail error', e));
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new function `helperSaveChannelDetail` is introduced to save channel details into the local database. It's a good practice to separate this logic into a helper function for better readability and maintainability. However, it would be better to handle errors in a more user-friendly way rather than just logging them to the console.
```diff
+ await Promise.all(promise).catch((e) => {
+   // Instead of just logging the error, consider showing an error message to the user or retrying the operation
+   console.log('saveChannelDetail error', e)
+ });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->